service/dps/storm/common/src/main/java/eu/europeana/cloud/service/dps/storm/spouts/kafka/QueueFiller.java

```diff
@@ -0,0 +1,72 @@
+package eu.europeana.cloud.service.dps.storm.spouts.kafka;
+
+import com.rits.cloning.Cloner;
+import eu.europeana.cloud.common.model.Representation;
+import eu.europeana.cloud.common.model.dps.States;
+import eu.europeana.cloud.mcs.driver.FileServiceClient;
+import eu.europeana.cloud.service.dps.PluginParameterKeys;
+import eu.europeana.cloud.service.dps.storm.NotificationTuple;
+import eu.europeana.cloud.service.dps.storm.StormTaskTuple;
+import eu.europeana.cloud.service.dps.storm.utils.TaskStatusChecker;
+import org.apache.storm.spout.SpoutOutputCollector;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.ArrayBlockingQueue;
+
+import static eu.europeana.cloud.service.dps.storm.AbstractDpsBolt.NOTIFICATION_STREAM_NAME;
+
+/**
+ * Created by Tarek on 9/14/2018.
+ */
+public class QueueFiller {
+    private TaskStatusChecker taskStatusChecker;
+    private SpoutOutputCollector collector;
+    ArrayBlockingQueue<StormTaskTuple> tuplesWithFileUrls;
+    private static final Logger LOGGER = LoggerFactory.getLogger(QueueFiller.class);
+
+     QueueFiller(TaskStatusChecker taskStatusChecker, SpoutOutputCollector collector, ArrayBlockingQueue<StormTaskTuple> tuplesWithFileUrls) {
+        this.taskStatusChecker = taskStatusChecker;
+        this.collector = collector;
+        this.tuplesWithFileUrls = tuplesWithFileUrls;
+    }
+
+     int addTupleToQueue(StormTaskTuple stormTaskTuple, FileServiceClient fileServiceClient, Representation representation) {
+        int count = 0;
+        final long taskId = stormTaskTuple.getTaskId();
+        if (representation != null) {
+            for (eu.europeana.cloud.common.model.File file : representation.getFiles()) {
+                String fileUrl = "";
+                if (!taskStatusChecker.hasKillFlag(taskId)) {
+                    try {
+                        fileUrl = fileServiceClient.getFileUri(representation.getCloudId(), representation.getRepresentationName(), representation.getVersion(), file.getFileName()).toString();
+                        final StormTaskTuple fileTuple = buildNextStormTuple(stormTaskTuple, fileUrl);
+                        tuplesWithFileUrls.put(fileTuple);
+                        count++;
+                    } catch (Exception e) {
+                        LOGGER.warn("Error while getting File URI from MCS {}", e.getMessage());
+                        count++;
```

Comment:
Why do you increase count when exception happens? Shouldn't it be increased only when the tuple was added to the queue?

Reply:
Yes we have to increase the number, if a record throws an exception we still have to count it :).
Example : you have an expected size of 5.
4 were handled without a problem and one record with an exception then we have to count it to so the actual size would be 5 and not 4 !.